inference-engine/thirdparty/clDNN/runtime/memory.cpp

```diff
@@ -20,26 +20,26 @@ namespace cldnn {
 memory::memory(engine* engine, const layout& layout, allocation_type type, bool reused)
     : _engine(engine), _layout(layout), _bytes_count(_layout.bytes_count()), _type(type), _reused(reused) {
     if (!_reused && _engine) {
-        _engine->add_memory_used(_bytes_count);
+        _engine->add_memory_used(_bytes_count, type);
     }
 
     GPU_DEBUG_GET_INSTANCE(debug_config);
     GPU_DEBUG_IF(debug_config->verbose >= 1) {
         GPU_DEBUG_COUT << "Allocate " << _bytes_count << " bytes of " << type << " allocation type"
-                       << " (current=" << _engine->get_used_device_memory() << ";"
+                       << " (current=" << _engine->get_used_device_memory(type) << ";"
                        << " max=" << _engine->get_max_used_device_memory() << ")" << std::endl;
     }
 }
 
 memory::~memory() {
     if (!_reused && _engine) {
-        _engine->subtract_memory_used(_bytes_count);
+        _engine->subtract_memory_used(_bytes_count, _type);
     }
 
     GPU_DEBUG_GET_INSTANCE(debug_config);
     GPU_DEBUG_IF(debug_config->verbose >= 1) {
-        GPU_DEBUG_COUT << "Free " << _bytes_count << " bytes"
-                       << " (current=" << _engine->get_used_device_memory() << ";"
+        GPU_DEBUG_COUT << "Free " << _bytes_count << " bytes of " << _type << " allocation type"
+                       << " (current=" << _engine->get_used_device_memory(_type) << ";"
```

Comment:
get_max_used_device_memory should be done for each alloc_type too

Reply:
done